src/hotspot/share/runtime/thread.cpp

```diff
@@ -2907,6 +2950,11 @@ void Threads::threads_do(ThreadClosure* tc) {
   non_java_threads_do(tc);
 }
 
+void Threads::threads_do(JavaThreadIteratorWithHandle* jtiwh_p, ThreadClosure* tc) {
+  java_threads_do(jtiwh_p, tc);
+  non_java_threads_do(tc);
```

Comment:
You should still have

```assert_locked_or_safepoint(Threads_lock);```

for non-JavaThread access.

Reply:
I'm backing out both:
```
Threads::java_threads_do(JavaThreadIteratorWithHandle* jtiwh_p, ThreadClosure* tc)
Threads::threads_do(JavaThreadIteratorWithHandle* jtiwh_p, ThreadClosure* tc)
```

and I'm backing out the changes to src/hotspot/share/services/management.cpp.
By removing the Threads_lock usage, I'm allowing those closures to run in parallel
with other closures that are using the original Threads::java_threads_do() and
Threads::threads_do() calls that do use the Threads_lock. Those other uses might
be assuming exclusivity for their algorithms and my M&M changes break that.